.changeset/shaggy-olives-melt.md

```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/model-generator': patch
+---
+
+respect --statement-typename-introspection and --statement-typename-introspection when generating types
```

Comment:
I'm assuming this should be `--statement-typename-introspection and --max-depth`?